### PR TITLE
Provide ability to tag GCE instances

### PR DESCRIPTION
**Pull request by krishnanrs**
GCE provides ability to tag instances and subsequently these tags
can be used in the network/firewall rules. Hence firewall rules
can be specified based on specific instance tags, which can be
assigned at launch time to control the network communication(s)
required by the specific GCE instance.
Online docs hace also been updated to reflect the new arguments/
usage.

### CODE REVIEW

**Comment by krishnanrs** [comment]
Couple of items of note:
1. The `validate_image_name` is now overloaded since the same validation applies for tags as well as instance names
2. No specific tests have been added

If there are any feedback/suggestions on the above items, let me know.

**Comment by krishnanrs** [comment]
Please take a look again.

File: brkt_cli/aws/diag_args.py [hunk 1]
```diff
@@ -67,7 +67,7 @@ def setup_diag_args(parser):
         help='Launch instances in this subnet'
     )
     parser.add_argument(
-        '--tag',
```
Comment (bburtin) [inline brkt_cli/aws/diag_args.py]: We should hide the original option and log deprecation warnings when they're used.  Maybe do this in a separate commit?

File: brkt_cli/gce/encrypt_gce_image_args.py [hunk 2]
```diff
@@ -72,6 +72,16 @@ def setup_encrypt_gce_image_args(parser, parsed_config):
         default=parsed_config.get_option('gce.subnetwork', None),
         required=False
     )
+    parser.add_argument(
+        '--gce-tag',
+        metavar='VALUE',
+        dest='tags',
```
Comment (bburtin) [inline brkt_cli/gce/encrypt_gce_image_args.py]: Can we call the arg field `gce_tags`, so that there isn't a discrepancy between the option name and variable name?

File: brkt_cli/gce/encrypt_gce_image_args.py [hunk 3]
```diff
@@ -72,6 +72,16 @@ def setup_encrypt_gce_image_args(parser, parsed_config):
...
+        action='append',
+        help=(
+              'Set a tag on the GCE encryptor instance. May be specified'
```
Comment (bburtin) [inline brkt_cli/gce/encrypt_gce_image_args.py]: The wording doesn't distinguish Bracket tags vs. CSP tags.  And this PR reignited the terminology debate.  Let's wait until the terminology is finalized before merging.

File: brkt_cli/gce/encrypt_gce_image_args.py [hunk 4]
```diff
@@ -110,6 +120,6 @@ def setup_encrypt_gce_image_args(parser, parsed_config):
         dest='crypto',
         metavar='NAME',
         choices=[CRYPTO_GCM, CRYPTO_XTS],
-        help=argparse.SUPPRESS,
-        default=None
+        help='Specify the name of the generated encrypted image',
```
Comment (bburtin) [inline brkt_cli/gce/encrypt_gce_image_args.py]: Can we put this into a separate commit?  Seems like it's not related to tags.

Comment (krishnanrs) [inline brkt_cli/gce/encrypt_gce_image_args.py]: I don't know why/how I got this change in. Will remove this from the commit.

File: brkt_cli/aws/diag_args.py (hunk 1 above)
Comment (krishnanrs) [inline brkt_cli/aws/diag_args.py]: If a user decides to use both the old and new option (just for kicks), should we print the warning and merge the two list of argument values?

**Comment by krishnanrs** [comment]
Please take a look again, pending agreement of the naming of tags vs claims or anything else.

**Comment by krishnanrs** [comment]
@bburtin can you take one final look?

File: brkt_cli/aws/__init__.py [hunk 5]
```diff
@@ -130,7 +130,17 @@ def run_diag(values):
...
     default_tags = {}
-    default_tags.update(brkt_cli.parse_tags(values.tags))
+    aws_tags = None
+    if values.tags:
+        log.warn(
+            'The "--tag" argument has been deprecated. Please use '
+            '"--aws-tag" instead.'
+        )
+        if values.aws_tags:
+            aws_tags = values.tags + values.aws_tags
```
Comment (bburtin) [inline brkt_cli/aws/__init__.py]: Since `--aws-tags` is replacing `--tags`, I think it's a bit magical for us to be appending the two lists.

File: brkt_cli/aws/__init__.py [hunk 6]
```diff
@@ -311,7 +331,17 @@ def run_update(values, config, verbose=False):
     encrypted_image = _validate_ami(aws_svc, values.ami)
     encryptor_ami = values.encryptor_ami or _get_encryptor_ami(values.region)
     default_tags = encrypt_ami.get_default_tags(nonce, encryptor_ami)
-    default_tags.update(brkt_cli.parse_tags(values.tags))
+    aws_tags = None
```
Comment (bburtin) [inline brkt_cli/aws/__init__.py]: Can we turn this into a single `get_aws_tags()` function instead of duplicating the code?

File: brkt_cli/aws/diag_args.py (hunk 1 above)
Comment (bburtin) [inline brkt_cli/aws/diag_args.py]: Sorry, I missed your original question about merging.

**Comment by krishnanrs** [comment]
Removed all AWS changes, except for minor doc updates. Will submit the AWS changes in a separate commit. Please review just for the GCE specific changes.
